### PR TITLE
Feature/edit task view

### PR DESCRIPTION
**Pull request by lmash**
### ISSUE/TICKET NUMBER
Create an edit Task View, Form & Template

### Describe the changes
Add edit task view
Link task list entries to a task in edit mode
Save tasks after editing

### Screenshots (if appropriate)
 - Before
 - After
task list with links
- ocd action

task in edit mode
- ocd action edit task

### Questions or comments (if any)
Add "n/a" if nothing to say

### CODE REVIEW

File: ocdaction/ocdaction/templates/dashboard/act/task_edit.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{% extends "layout.html" %}
+
+{% load static %}
+
+{% block main %}
+    <section>
+        <h1>Edit task</h1>
+        <form class="col-xs-6 col-xs-offset-3" method="POST" action="{{ task_form.get_absolute_url }}">
+            {% csrf_token %}
```
Comment (lili2311) [inline ocdaction/ocdaction/templates/dashboard/act/task_edit.html]: Please could you convert this form to use widget tweaks? Example here it is mostly copy paste :D https://github.com/womenhackfornonprofits/ocdaction/blob/904134b35599e0d7f03b97db7b9280236d2c803e/ocdaction/ocdaction/templates/profiles/login.html

File: ocdaction/tasks/models.py [hunk 2]
```diff
@@ -12,11 +13,16 @@ class Task(models.Model):
     task_fears = models.CharField(max_length=300, blank=True)
     task_compulsions = models.CharField(max_length=300, blank=True)
```
Comment (lili2311) [inline ocdaction/tasks/models.py]: Might need to add `widget=forms.Textarea)` since the field is not showing up as a textarea.

**Comment by lili2311** [comment]
Looking great! Left a few comments also please resolve the conflict you have an let's get this merged :)

**Comment by lmash** [comment]
@lili2311 suggested changes made - hopefully ok now

**Comment by lili2311** [comment]
Thanks those are great but I noticed the fields are not textareas, can we add the widget to each one to render as text areas as one line is not probably enough to describe fears etc?

**Comment by lmash** [comment]
@lili2311 for the textarea change - can this be done using widget tweaks? If so have you an example.. I'm not having much success so far

**Comment by lili2311** [comment]
You need to set the widget type you want in the Form https://docs.djangoproject.com/en/1.10/ref/forms/widgets/ you can only change input type in django tweaks not to a dfferent element.

So in the form just override the field with a widget type:
  ```comment = forms.CharField(widget=forms.Textarea)```

**Comment by lmash** [comment]
thanks @lili2311 makes more sense now.. I've set rows = 3 for task_fears, task_compulsions and task_goals

File: ocdaction/ocdaction/dashboard_urls.py [hunk 3]
```diff
@@ -22,6 +23,11 @@
         name="task-add"
     ),
     url(
+        r'^tasks/(?P<task_id>\d+)/$',
```
Comment (lili2311) [inline ocdaction/ocdaction/dashboard_urls.py]: can we change the url to have edit in it as we would also want to be able to view a task on this url. So if we can have the `^tasks/(?P<task_id>\d+)/edit$` for this view then the `^tasks/(?P<task_id>\d+)/$` is available for us for later to be able to view a task without editing

**Pull request merged**